aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -185,6 +208,40 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+
+    public entry fun rotate_authentication_key_ed25519(account: &signer, signature: vector<u8>, new_public_key: vector<u8>) acquires Account, OriginatingAddress {
+        let addr = signer::address_of(account);
+        assert!(exists_at(addr), error::not_found(EACCOUNT));
+        assert!(
+            vector::length(&new_public_key) == 32,
+            error::invalid_argument(EMALFORMED_PUBLIC_KEY)
+        );
+        assert!(
+            vector::length(&signature) == 64,
+            error::invalid_argument(EMALFORMED_PROOF_OF_KNOWLEDGE)
+        );
+
+        let account_resource = borrow_global_mut<Account>(addr);
+        let current_auth_key = create_address(account_resource.authentication_key);
```

Comment:
I would expect `create_address` to return an address not an authentication key. I realize they are the same in the current design, but this is confusing. What is going on here?

Reply:
I guess you are just trying to fetch the current auth key at this address and store it in `proof` which (confusingly) uses `address` as the type of an auth key?

Could we have some basic type safety here?

Reply:
Actually, I'm confused why isn't this just `current_auth_key = account_resource.authentication_key`?

Reply:
I think the address and auth key are the same in the current design, except that address is of type `address` and current auth key is of type `vector<u8>`. we wanted to store <address, address> in the OriginatingAddress map so we're doing `create_address` on the auth key here